cpp/test/sg/dbscan_test.cu

```diff
@@ -46,72 +61,79 @@ template <typename T>
 class DbscanTest : public ::testing::TestWithParam<DbscanInputs<T>> {
  protected:
   void basicTest() {
-    cudaStream_t stream;
-    CUDA_CHECK(cudaStreamCreate(&stream));
+    cumlHandle handle;
 
     params = ::testing::TestWithParam<DbscanInputs<T>>::GetParam();
-    Random::Rng r(params.seed);
-    int len = params.len;
 
-    allocate(data, len);
+    device_buffer<T> out(handle.getDeviceAllocator(), handle.getStream(),
+                         params.n_row * params.n_col);
+    device_buffer<int> l(handle.getDeviceAllocator(), handle.getStream(),
+                         params.n_row);
 
-    std::vector<T> data_h = {1.0, 2.0, 2.0, 2.0, 2.0,  3.0,
-                             8.0, 7.0, 8.0, 8.0, 25.0, 80.0};
-    data_h.resize(len);
-    updateDevice(data, data_h.data(), len, stream);
+    make_blobs(handle, out.data(), l.data(), params.n_row, params.n_col,
+               params.n_centers, nullptr, nullptr, params.cluster_std, true,
+               -10.0f, 10.0f, 1234ULL, false);
 
     allocate(labels, params.n_row);
     allocate(labels_ref, params.n_row);
-    std::vector<int> labels_ref_h = {0, 0, 0, 1, 1, -1};
-    labels_ref_h.resize(len);
-    updateDevice(labels_ref, labels_ref_h.data(), params.n_row, stream);
 
-    T eps = 3.0;
-    int min_pts = 2;
-    cumlHandle handle;
-    handle.setStream(stream);
+    MLCommon::copy(labels_ref, l.data(), params.n_row, handle.getStream());
+
+    CUDA_CHECK(cudaStreamSynchronize(handle.getStream()));
+
+    dbscanFit(handle, out.data(), params.n_row, params.n_col, params.eps,
+              params.min_pts, labels, params.max_bytes_per_batch, false);
 
-    // forces a batch size of 2
-    size_t max_bytes_per_batch = 4 * sizeof(T) * 8;
+    CUDA_CHECK(cudaStreamSynchronize(handle.getStream()));
 
-    dbscanFit(handle, data, params.n_row, params.n_col, eps, min_pts, labels,
-              max_bytes_per_batch, true);
-    CUDA_CHECK(cudaStreamSynchronize(stream));
+    score = adjustedRandIndex(handle, labels_ref, labels, params.n_row, 0,
```

Comment:
This is nice! We finally have a way to verify our algos based on some metrics inside our unit-tests! :)